sklearn/decomposition/tests/test_incremental_pca.py

```diff
@@ -384,25 +384,38 @@ def test_singular_values():
     assert_array_almost_equal(ipca.singular_values_, [3.142, 2.718, 1.0], 14)
 
 
-def test_whitening():
+def test_whitening(global_random_seed):
     # Test that PCA and IncrementalPCA transforms match to sign flip.
     X = datasets.make_low_rank_matrix(
-        1000, 10, tail_strength=0.0, effective_rank=2, random_state=1999
+        1000, 10, tail_strength=0.0, effective_rank=2, random_state=global_random_seed
     )
-    prec = 3
-    n_samples, n_features = X.shape
+    atol = 1e-3
     for nc in [None, 9]:
         pca = PCA(whiten=True, n_components=nc).fit(X)
         ipca = IncrementalPCA(whiten=True, n_components=nc, batch_size=250).fit(X)
 
+        # Since the data is rank deficient, some components are pure noise. We
+        # should not expect those dimensions to carry any signal and their
+        # values might be arbitrarily changed by implementation details of the
+        # internal SVD solver. We therefore mask them out before comparison.
```

Comment:
Do we want to test rank deficient data or could we simplify the test with other data X?

Reply:
I think the test is stronger this way. `IncrementalPCA` is typically used when `n_samples >> n_features` and with the default value of `n_components=None` this case is bound to happen in practice.

Reply:
Note that `test_pca_solver_equivalence` covers even more cases and actually found a bug in for a related case with `PCA(whiten=True, svd_solver="arpack")` (the one documented in the changelog), so it's actually useful to (also) test with rank deficient data.